### ISSUE TITLE
Support default colors for dark theme

### ISSUE THREAD

**Issue by de-jcup** [boothen/Json-Eclipse-Plugin#24]
The default font colors are suitable for light theme but there seems to be no defaults for dark theme. When installing the plugin fresh from marketplace to a dark theme eclipse (4.11) it looks very bad:

![image](https://user-images.githubusercontent.com/11644753/52218825-ba2c0600-289b-11e9-8288-804c051cf265.png)

I would suggest to provide either a dedicated defaults button in preferences or to provide a css file for dark theme support (e4 style).

Here my suggestions for a good looking defaults:

![image](https://user-images.githubusercontent.com/11644753/52218347-c6fc2a00-289a-11e9-9088-1bfb1d322969.png)

![image](https://user-images.githubusercontent.com/11644753/52218963-fe1f0b00-289b-11e9-908f-022e6663966e.png)

**Comment by msteiger** [comment]
How do you configure the color theme? Do you use the eclipse theme plugin?

If so, it looks like a duplicate of #21 - can you confirm?

**Comment by de-jcup** [comment]
It seems #21 is some sort of duplicate her. But with some differences:
- I propose some default colors
- I would appreciate dedicated dark theme coloring done by json plugin itself.
  Some defaults colors are okay to reuse - e.g. when it comes to hyperlinking... but some 
  colors shall be defined in plugins itself

The eclipse theme plugin is dead - as you mentioned in #21 

But eclipse 4.x provides its own theme mechanism done by css. 
Please look at 
https://github.com/de-jcup/eclipse-asciidoctor-editor/pull/63/commits/be24fb4c87b964edef48613c561f1bb8e031c087
where Lars Vogel did provide a css template for asciidoctor editor.

I think you can use this commit as working example.

**Comment by de-jcup** [comment]
For common defaults you can also reuse current colors by code see
https://github.com/de-jcup/eclipse-asciidoctor-editor/issues/70

> We had this in the past also in Platform but fixed that with https://bugs.eclipse.org/bugs/show_bug.cgi?id=517393
> I think you should be able to get the correct color via:
> 
> JFaceResources.getColorRegistry().JFacePreferences.HYPERLINK_COLOR);
> JFaceResources.getColorRegistry().JFacePreferences.ACTIVE_HYPERLINK_COLOR);

**Comment by de-jcup** [comment]
Be awarea about my comment at https://github.com/de-jcup/eclipse-asciidoctor-editor/issues/174#issuecomment-462060474 and how the css setup is applied...

**Comment by msteiger** [comment]
Thanks for the hint. I will dig it into, but might need a few days for that.

**Comment by msteiger** [comment]
I've spent a bit of time on this and configured the CSS for the dark theme, but it doesn't seem to get picked up by the preference store. https://github.com/boothen/Json-Eclipse-Plugin/commit/ac8d179a7d935af905bdeec73f0b54f75c7e6194

Is the name of the "pseudo attribute" important? The [JsonTokenStyler](https://github.com/boothen/Json-Eclipse-Plugin/blob/master/jsonedit-preferences/src/main/java/com/boothen/jsonedit/preferences/JsonTokenStyler.java#L37) uses the PreferenceStore of the [TextSourceViewerConfiguration](https://github.com/boothen/Json-Eclipse-Plugin/blob/master/jsonedit-editor/src/main/java/com/boothen/jsonedit/editor/JsonSourceViewerConfiguration.java#L63) to get the colors.

Maybe my minimal runtime is too old? It currently uses org.eclipse.ui.editors_3.10.0.v20160505  and org.eclipse.e4.ui.css.swt.theme_0.10.100.v20160523 ...

**Comment by de-jcup** [comment]
I have also some issues with the new theme. Sometimes it works sometimes not... a little bit frustrating...
see [asciidoctor-editor#185](https://github.com/de-jcup/eclipse-asciidoctor-editor/issues/185)

Also the "Restore defaults" button in preferences works not correctly when dark theme is activate... seems to be an older and wellknown bug in eclipse: https://bugs.eclipse.org/bugs/show_bug.cgi?id=478835

I end up with supporting a "Restore defaults for dark theme" button - see https://github.com/de-jcup/eclipse-bash-editor/wiki/FAQ#how-to-apply-default-dark-theme-colours for an example.

I will try to provide the theme color css as well, so if this is fixed/initial the colors should automatically apepar.

So my suggestion would be to add a "Restore defaults for dark theme" button in your preferences ...
Works always.

**Comment by PyvesB** [comment]
The defaults do work very badly with the dark theme. Just hit this problem today.

**Comment by fkrivsky** [comment]
This issue is still present...

**Comment by msteiger** [comment]
Yes, I could not find a good way to fix this. The eclipse default color support does not work as I expected and I couldn't find proper documentation or examples for it. Even with the ascii-doctor source code, I could not get it to work properly. If someone knows how to do it, a PR would be highly appreciated.
